create_bulk_tasks.py

```diff
@@ -0,0 +1,140 @@
+
+handler = logging.StreamHandler(sys.stdout)
+handler.setLevel(logging.DEBUG)
+handler.setFormatter(formatter)
+root_logger.addHandler(handler)
+logger = logging.getLogger("datapurge")
+
+
+def create_task(project_id, created, num_answers, priority, year):
+    sql = """
+        INSERT INTO task
+            (created, project_id, state, priority_0, info, n_answers, exported, quorum, calibration)
+        VALUES
+            (%(created)s, %(project_id)s, %(state)s, %(priority_0)s, %(info)s, %(n_answers)s, %(exported)s, %(quorum)s, %(calibration)s)
+        RETURNING id;
+    """
+    columns_values = dict(
+        created=created,
+        project_id=project_id, state="complete",
+        info=json.dumps({"company": "bberg", "earning_year": year}),
+        exported=True, n_answers=num_answers, priority_0=priority,
+        quorum=0, calibration=0
+    )
+    try:
+        with AccessDatabase() as db:
+            db.execute_sql(sql, columns_values)
+            db.conn.commit()
+            return db.cursor.fetchone()[0]
+    except (Exception, psycopg2.DatabaseError) as error:
+        logger.exception("Error: ", error)
+
+def create_task_run(project_id, task_id, user_id, created):
+    sql = """
+        INSERT INTO task_run
+            (created, project_id, task_id, user_id, info, finish_time, timeout, calibration)
+        VALUES
+            (%(created)s, %(project_id)s, %(task_id)s, %(user_id)s, %(info)s,
+            %(finish_time)s, %(timeout)s, %(calibration)s)
+        RETURNING id;
+    """
+    year = randrange(2018, 2022)
+
+    columns_values = dict(
+        created=created,
+        project_id=project_id, task_id=task_id,
+        info=json.dumps({"company": "Bloomberg", "earning_year": year}),
+        user_id=user_id, finish_time=created, timeout=30, calibration=0
+    )
+    try:
+        with AccessDatabase() as db:
+            db.execute_sql(sql, columns_values)
+            db.conn.commit()
+            return db.cursor.fetchone()[0]
+    except (Exception, psycopg2.DatabaseError) as error:
+        logger.exception("Error: ", error)
+
+def create_result(project_id, task_id, task_run_ids, created):
+    sql = """
+        INSERT INTO result
+            (created, project_id, task_id, task_run_ids, info)
+        VALUES
+            (%(created)s, %(project_id)s, %(task_id)s, %(task_run_ids)s, %(info)s)
+        RETURNING id;
+    """
+    year = randrange(2018, 2022)
+
+    columns_values = dict(
+        created=created,
+        project_id=project_id, task_id=task_id,
+        task_run_ids=task_run_ids,
+        info=json.dumps({"company": "Bloomberg", "earning_year": year})
+    )
+    try:
+        with AccessDatabase() as db:
+            db.execute_sql(sql, columns_values)
+            db.conn.commit()
+            return db.cursor.fetchone()[0]
+    except (Exception, psycopg2.DatabaseError) as error:
+        logger.exception("Error: ", error)
+
+
+def create_bulk_data(project_id, num_records):
```

Comment:
why do we need to randomly generate data?

Reply:
this is to simulate real life situation wherein task submissions can be done at different times. it is useful in cases where we want to use the test data of different data; an example testing purging task data older than 12 months or so requires bulk tasks created of older date as well as current date so that only old data would be deleted.